Documentation/api-reference.md

```diff
@@ -68,72 +73,187 @@ m.aepSdk = AdobeAEPSDKInit()
+
+| Constants | Raw value | Type | Required |
+| :-- | :--: | :--: | :--: |
+| `ADB_CONSTANTS.CONFIGURATION.MEDIA_CHANNEL` | "edgemedia.channel" | String | **Yes**
+| `ADB_CONSTANTS.CONFIGURATION.MEDIA_PLAYER_NAME` | "edgemedia.playerName" | String | **Yes**
+| `ADB_CONSTANTS.CONFIGURATION.MEDIA_APP_VERSION` | "edgemedia.appVersion" | String | **No**
+
+##### Syntax
+
+```brightscript
+updateConfiguration: function(configuration as object) as void
+```
+
+- `@param configuration as object`
+
+##### Example
+
+```brightscript
+ADB_CONSTANTS = AdobeAEPSDKConstants()
+
+configuration = {}
+configuration[ADB_CONSTANTS.CONFIGURATION.EDGE_CONFIG_ID] = "<YOUR_CONFIG_ID>"
+configuration[ADB_CONSTANTS.CONFIGURATION.EDGE_DOMAIN] = "<YOUR_DOMAIN_NAME>"
+
+' Set the default consent to pending and update the consent later based on user preference
+configuration[ADB_CONSTANTS.CONFIGURATION.CONSENT_DEFAULT] = {
+    "consents": {
+        "collect": {
+            "val": "p"
+        }
+    }
+}
+
+m.aepSdk.updateConfiguration(configuration)
+```
+
+The `EDGE_CONFIG_ID` value is presented as `Datastream ID` in the [Datastream details](https://experienceleague.adobe.com/docs/experience-platform/edge/datastreams/configure.html?lang=en#view-details) page.
+
+The `EDGE_DOMAIN` value is the first-party domain mapped to the Adobe-provisioned Edge Network domain. For more information, see this [documentation](https://developer.adobe.com/client-sdks/documentation/edge-network/#domain-configuration)
+
+### Configure default consent
+
+The default consent configuration determines how data collection consent is managed before invoking the [setConsent](#setconsent) API. This setting is particularly important to avoid unintentionally collecting data from users in regions where consent is required prior to data collection.
+
+By default, users are opted in for all purposes.
+
+#### Example: Setting default collect consent to pending before collecting actual user consent
+```brightscript
+ADB_CONSTANTS = AdobeAEPSDKConstants()
+
+configuration = {}
+
+configuration[ADB_CONSTANTS.CONFIGURATION.CONSENT_DEFAULT] = {
+    "consents": {
+        "collect": {
+            "val": "p"
+        }
+    }
+}
+
+m.aepSdk.updateConfiguration(configuration)
+```
+
+## Consent APIs:
+
+### setConsent
+
+Sends consent preferences to the Edge Network. For details on setting default consent before collecting user preferences, refer to the [Configure default consent](#configure-default-consent) section.
+
+> [!Important]
+> Please provide the entire payload with all the fields required by the Adobe 2.0 Standard.
+
+##### Syntax
+
+```brightscript
+setConsent: function(data as object) as void
+```
+
+- `@param data as object : an associative array that includes consent data to be sent to the edge network.`
+
+##### Example: setConsent using consent object following Adobe 2.0 Standard
+
+> [!Note]
+> The "time" field under "metadata" should contain timestamp in ISO format as shown below.
```

Comment:
should contain ISO 8601 formatted with UTC(Z) time zone, millisecond precision date